modules/cloudflarepurger/src/Plugin/Purge/DiagnosticCheck/ApiRateLimitCheck.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\cloudflarepurger\Plugin\Purge\DiagnosticCheck\ApiRateLimitCheck.
+ */
+
+namespace Drupal\cloudflarepurger\Plugin\Purge\DiagnosticCheck;
+
+use Drupal\purge\Plugin\Purge\DiagnosticCheck\DiagnosticCheckBase;
+use Drupal\purge\Plugin\Purge\DiagnosticCheck\DiagnosticCheckInterface;
+use Drupal\cloudflare\CloudFlareStateInterface;
+use CloudFlarePhpSdk\ApiEndpoints\CloudFlareAPI;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Checks that the site is within CloudFlare's API rate limits.
+ *
+ * CloudFlare currently has a rate limit of 1200 Api calls every 5 minutes.
```

Comment:
Nitpick: perhaps a ``@todo`` just to assure that everyone understands that you want this (limit) gone at some point?